proposals/manegiable-aff-data-mapping/proposal.md

```diff
@@ -0,0 +1,243 @@
+- **Role:** Establish **scoping**, and define **relationships** between interactions.
+
+---
+
+The idea is that a Capability can define the following:
+
+- **State (set of Variables):** Define runtime client data structure shared by all affordances listed in it (e.g., `activeJobQueue`).
+- **Subaffordances:** Group related Actions, Properties, and Events that operate within the same context.
+- **Model**: Without needing to define all the affordances from scratch every time, we can define a set of common Thing Models (e.g., AsyncActionThingModel).
+
+Example:
+
+```json
+{
+  "capabilities": {
+    "PrintJobManager": {
+      "state": {
+        "type": "string",
+        "variable": "jobId"
+      },
+      "properties": {
+        "activeJobQueue": {
+          "type": "array",
+          "items": { "type": "string" },
+          "forms": [
+            {
+              "href": "http://example.com/printjobs/queue",
+              "htv:methodName": "GET"
+            }
+          ]
+        }
+      },
+      "actions": {
+        "submitPrintJob": {
+          "input": {
+            "type": "object",
+            "properties": {
+              "document": { "type": "string", "format": "uri" },
+              "copies": { "type": "integer", "default": 1 }
+            }
+          },
+          "output": {
+            "type": "string",
+            "variable": "jobId"
+          },
+          "forms": [
+            {
+              "href": "http://example.com/printjobs",
+              "htv:methodName": "POST"
+            }
+          ]
+        },
+        "cancelPrintJob": {
+          "input": {
+            "type": "string",
+            "variable": "jobId"
+          },
+          "forms": [
+            {
+              "href": "http://example.com/printjobs/${jobId}",
+              "htv:methodName": "DELETE"
+            }
+          ]
+        }
+      },
+      "events": {
+        "jobCompleted": {
+          "data": {
+            "type": "string",
+            "variable": "jobId"
+          },
+          "forms": [
+            {
+              "href": "http://example.com/printjobs/events"
+            }
+          ]
+        }
+      }
+    }
+  }
+}
+```
+
+We can simplify by defining a common Thing Model for such Capabilities (e.g., `PrintJobManagerThingModel`) that pre-defines the common affordances and state variables. The TD can then just reference this model and only define any customizations needed.
+Example:
+
+```json
+{
+  "capabilities": {
+    "PrintJobManager": {
+      "model": "http://example.com/PrintJobManagerThingModel"
+      // Customizations or overrides can be defined here if needed
+    }
+  }
+}
+```
+
+#### Limitations & Future Work
+
+- Naming and Terminology: The term "Capability" may conflict with existing IoT frameworks. Future work may explore alternative names like "Cluster" or "Module" or "Group" "RuntimeContext".
```

Comment:
It already conflicts with our terminology where we use capability together with affordance. Searching for "capabilit" in https://www.w3.org/TR/wot-architecture11 results in this. Not very happy with the current state but probably it is in people's minds already.